.github/workflows/update-rpc.yml

```diff
@@ -0,0 +1,94 @@
+name: Update Submodule
+
+# on:
+#   release:
+#     types: [published]
+
+on: [pull_request]
+
+permissions: write-all
+
+jobs:
+  update-submodule:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Kakarot Repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Get Release Tag
+        id: get_release_tag
+        # run: echo "release_tag=${{ github.event.release.tag_name }}" >> $GITHUB_ENV
+        run: echo "release_tag=v0.8.2" >> $GITHUB_ENV
+
+      - name: Checkout Kakarot-RPC Repository
+        uses: actions/checkout@v4
+        with:
+          repository: kkrt-labs/kakarot-rpc
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: kakarot-rpc
+
+      - name: Check Current Submodule Commit
+        id: check_submodule_commit
+        run: |
+          cd kakarot-rpc/lib/kakarot
+          current_commit=$(git rev-parse HEAD)
+          latest_commit=$(git ls-remote https://github.com/kkrt-labs/kakarot.git ${{ env.release_tag }} | awk '{print $1}')
+          echo "current_commit=${current_commit}" >> $GITHUB_ENV
+          echo "latest_commit=${latest_commit}" >> $GITHUB_ENV
+          cd -
+
+      - name: Check if Branch Exists
+        id: check_branch_exists
+        run: |
+          cd kakarot-rpc
+          if git ls-remote --exit-code --heads origin kakarot-${{ env.latest_commit }}; then
+            echo "branch_exists=true" >> $GITHUB_ENV
+          else
+            echo "branch_exists=false" >> $GITHUB_ENV
+          fi
+          cd -
```

Comment:
doesn't seem necessary to me, why is it?

Reply:
I was thinking of doing this as a safety measure so that if the branch already exists on the RPC repo for the same commit update for some reason, then we don't do a second equivalent PR.